CodePush.m

```diff
@@ -362,11 +366,19 @@ - (void)savePendingUpdate:(NSString *)packageHash
 /*
  * This is native-side of the RemotePackage.download method
  */
-RCT_EXPORT_METHOD(downloadUpdate:(NSDictionary*)updatePackage
+RCT_EXPORT_METHOD(downloadUpdate:(NSDictionary*)immutableUpdatePackage
                         resolver:(RCTPromiseResolveBlock)resolve
                         rejecter:(RCTPromiseRejectBlock)reject)
 {
     dispatch_async(dispatch_get_main_queue(), ^{
+        NSDictionary* updatePackage = [immutableUpdatePackage mutableCopy];
+        NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
```

Comment:
Ah right, that's a good point. I guess this is yet another example of why our current API has some odd consequences as opposed to having a more intuitive entry point for the plugin than `bundleURL`.